internal/capsule/docker/docker.go

```diff
@@ -0,0 +1,46 @@
+package docker
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"github.com/caesium-dev/caesium/internal/capsule"
+	"github.com/docker/docker/api/types"
+	containertypes "github.com/docker/docker/api/types/container"
+	networktypes "github.com/docker/docker/api/types/network"
+	specs "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type dockerBackend interface {
+	ContainerInspect(context.Context, string) (types.ContainerJSON, error)
+	ContainerList(context.Context, types.ContainerListOptions) ([]types.Container, error)
+	ContainerCreate(context.Context, *containertypes.Config, *containertypes.HostConfig, *networktypes.NetworkingConfig, *specs.Platform, string) (containertypes.ContainerCreateCreatedBody, error)
+	ContainerStart(context.Context, string, types.ContainerStartOptions) error
+	ContainerStop(context.Context, string, *time.Duration) error
+	ContainerRemove(context.Context, string, types.ContainerRemoveOptions) error
+	ContainerLogs(context.Context, string, types.ContainerLogsOptions) (io.ReadCloser, error)
+}
+
+var (
+	stateMap = map[string]capsule.State{
+		"created":    capsule.Created,
+		"running":    capsule.Running,
+		"paused":     capsule.Invalid, // a container should never be paused
+		"restarting": capsule.Invalid, // a container should never be restarting
```

Comment:
A *capsule* should never be paused/restarting.

Reply:
Well yes, but this was in the docker package, and I was addressing _docker_ states.